Sources/ArcGISToolkit/Components/Offline/OnDemand/OnDemandMapAreaSelectorView.swift

```diff
@@ -0,0 +1,310 @@
+    /// - Parameter handle: The drag point handle.
+    /// - Returns: The minimum rect for a handle.
+    private func minimumRect(forHandle handle: DragPoint) -> CGRect {
+        let maxWidth: CGFloat = 50
+        let maxHeight: CGFloat  = 50
+        
+        switch handle {
+        case .topLeft:
+            // Anchor is opposite corner.
+            return CGRect(
+                x: boundingRect.maxX - maxWidth,
+                y: boundingRect.maxY - maxHeight,
+                width: maxWidth,
+                height: maxHeight
+            )
+        case .topRight:
+            return CGRect(
+                x: boundingRect.minX,
+                y: boundingRect.maxY - maxHeight,
+                width: maxWidth,
+                height: maxHeight
+            )
+        case .bottomLeft:
+            return CGRect(
+                x: boundingRect.maxX - maxWidth,
+                y: boundingRect.minY,
+                width: maxWidth,
+                height: maxHeight
+            )
+        case .bottomRight:
+            return CGRect(
+                x: boundingRect.minX,
+                y: boundingRect.minY,
+                width: maxWidth,
+                height: maxHeight
+            )
+        }
+    }
+    
+    /// Updates the handle locations using the boudning rect.
+    private func updateHandles() {
+        topRight = CGPoint(x: boundingRect.maxX, y: boundingRect.minY)
+        topLeft = CGPoint(x: boundingRect.minX, y: boundingRect.minY)
+        bottomLeft = CGPoint(x: boundingRect.minX, y: boundingRect.maxY)
+        bottomRight = CGPoint(x: boundingRect.maxX, y: boundingRect.maxY)
+    }
+    
+    /// The handle view for the map area selector.
+    struct Handle: View {
+        /// The position of the handle.
+        let position: CGPoint
+        /// The closure to call when the map area selector should be resized.
+        let resize: (CGPoint) -> Void
+        /// The gesture state of the drag gesture.
+        @GestureState var gestureState: State = .started
+        /// The types of gesture states.
+        enum State {
+            case started, changed
+        }
+        
+        var body: some View {
+            Color.clear
+                .contentShape(Rectangle())
+                .frame(width: 44, height: 44)
+                .position(position)
+                .gesture(DragGesture(coordinateSpace: .local)
+                    .updating($gestureState) { value, state, _ in
+                        switch state {
+                        case .started:
+                            state = .changed
+                            UISelectionFeedbackGenerator().selectionChanged()
+                        case .changed:
+                            resize(value.location)
+                        }
+                    }
+                )
+        }
+    }
+    
+    /// A view that displays rounded corners for a rectangle view.
+    struct RoundedCorners: Shape {
```

Comment:
It's fine to leave it like this for now, but I would've expected the Handle view to draw the rounded corner. So the handle view would have been initialized with a HandleLocation and it would draw the corner based on its location. Instead of the handle being a clear view. Maybe something for a future PR.